EvidenceApi/V1/Controllers/BaseController.cs

```diff
@@ -25,5 +30,55 @@ public static void ConfigureJsonSerializer()
                 return settings;
             };
         }
+
+        public override void OnActionExecuting(ActionExecutingContext filterContext)
+        {
+            var userEmail = filterContext.HttpContext.Request.Headers["UserEmail"];
+            if (userEmail.Count == 0)
+            {
+                filterContext.Result = new BadRequestObjectResult("UserEmail request header empty");
+                return;
+            }
+
+            var userEmailValue = userEmail[0];
+            if (SkipAuditForResidentRequest(userEmailValue))
+            {
+                base.OnActionExecuting(filterContext);
+                return;
+            }
+
+            var path = filterContext.HttpContext.Request.Path.Value;
+            var method = filterContext.HttpContext.Request.Method;
+            var request = "";
+
+            if (method == "POST" || method == "PATCH")
+            {
+                request = JsonConvert.SerializeObject(filterContext.ActionArguments["request"], Formatting.None);
+            }
+
+            var queryString = filterContext.HttpContext.Request.QueryString.Value;
```

Comment:
You could put this line within the `if (method == "GET")` because we only need to retrieve the queryString for GET requests

Reply:
👍